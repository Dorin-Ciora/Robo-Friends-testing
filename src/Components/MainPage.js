import React, { Component } from 'react';
import CardList from '../Components/CardList';
import SearchBox from '../Components/SearchBox';
import Scroll from '../Components/Scroll';
import './MainPage.css';
import Header from '../Components/Header';


class MainPage extends Component {
    componentDidMount () {
        this.props.onRequestRobots();
    }

    filterRobots = robots => {
        return robots.filter(robot => {
        return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
    })}

    render() {
         const { onSearchChange, robots, isPending } =this.props;
         return isPending ?
            <h1>Loading</h1> :
        (
            <div className='tc'>
           <Header />
            <SearchBox searchChange={onSearchChange}/>
            <Scroll>
            <CardList robots={this.filterRobots(robots)}/>
            </Scroll> 
            </div>
        );
        }
}
export default MainPage;