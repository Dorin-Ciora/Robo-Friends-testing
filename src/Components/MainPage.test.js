import { shallow } from 'enzyme';
import CounterButtons from './CounterButtons';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(()=> {
    const mockProps ={
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps} />)
})
it('renders MainPage without crash', ()=> {
expect(wrapper).toMatchSnapshot();
})

it('filter robots correctly', () => {
    expect(wrapper.instance().filterRobots([])).toEqual([])
})