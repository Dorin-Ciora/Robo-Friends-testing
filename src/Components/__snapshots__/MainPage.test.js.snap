// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crash 1`] = `ShallowWrapper {}`;

exports[`should render CounterButton component 1`] = `ShallowWrapper {}`;
